### DISCUSSION TITLE
promicroが認識しない

### DISCUSSION THREAD

**Discussion by curuknagao** [Yowkees/keyball#409] [Q&A]
当方自作キーボード初心者です。
先日keyball44を購入し、ProMicroまで取り付けしたところでテストファームウェアを流そうとusb-cで接続しましたが、promicroが赤く光るだけで、OLEDもLEDも何もつかず、remapからキーボードの認識がされないです。
PCがubuntuなので`lsusb`コマンドで見てみると、`Bus 001 Device 004: ID 1b4f:9205 SparkFun Electronics Pro Micro 5V`として認識はしているような気はします。
何をどうリカバーしたらいいでしょうか？

**Comment by megmilk** [comment]
@curuknagao さん

PCと接続したままリセットボタンを素早く2回押すと、その後数秒間だけファームウェアが書き込めるはずです

Ubuntu だと `/dev/hidraw***` を 一時的に chmod で `0666` にする必要があります
どうやら連番でデバイス3つぐらい使っています

Windows や macOS なら chmod は必要なかったです

リセットボタン2回からの chmod は、素早い操作が必要なので、ファームウェアの書き込みは Ubuntu 以外が良いと思います

**Reply by curuknagao** [reply]
ありがとうございます！firmwareは書き込めたようです！

oledにkeyballと表示されました。
ただいくつかキーが反応しないものがありました。
キーが反応しない場合の対応としてはキーソケットのハンダやり直しとダイオードのやり直しをしたほうがいいのでしょうか？

どのキーがどのダイオードに対応しているか不明なのでその辺り説明してるページなどあれば教えていただけると助かります。

**Reply by megmilk** [reply]
@curuknagao さん

> いくつかキーが反応しないものがありました。

テスト用のファームウェアは、修飾キーなどが配置されています

そのため、一見するとキーが効いていないように感じることがあります

ただ、私はハッキリと場所を覚えておりません

> どのキーがどのダイオードに対応しているか不明なのでその辺り説明してるページなどあれば教えていただけると助かります。

このあたりはヨーキースさんが詳しいはずです

お力になれずスミマセン

**Reply by curuknagao** [reply]
いえいえ、ありがとうございます！

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。またmegmilk様アドバイスありがとうございます。

一部入力できないキーについてはRemapのキーマトリクス機能でご確認いただき、
キーソケットと周辺のダイオードの半田付けを見直してみて下さい。
どうしても解決しないキーがありましたら半田付け面のお写真を見せていただければアドバイスできると思います。

以上、よろしくお願いいたします。
